docs/src/examples/ode/freefall.py

```diff
@@ -0,0 +1,145 @@
+
+def onroot_vb(t, y, solver):
+    return 1
+
+def onroot_vc(t, y, solver):
+    if t > 28: # we have found 4 interruption points, so we stop
+        return 1
+
+    solver.reinit_IC(t, [Y0, y[1]])
+
+    return 0
+
+
+def print_results(experiment_no, result, require_no_roots=False):
+    ts, ys = result.values.t, result.values.y
+    # Print computed values at tspan
+    print('\n Experiment number: ', experiment_no)
+    print('--------------------------------------')
+    print('    t             Y               v')
+    print('--------------------------------------')
+
+    for (t, y, v) in zip(ts, ys[:, 0], ys[:, 1]):
+        print('%6.1f %15.4f %15.2f' % (t, y, v))
+
+
+    t_roots, y_roots = result.roots.t, result.roots.y
+    if not require_no_roots:
+        # Print interruption points
+        print('\n t_roots     y_roots        v_roots')
+        print('--------------------------------------')
+        if (t_roots is None) and (y_roots is None):
+            print('%6s %15s %15s' % (t_roots, y_roots, y_roots))
+        elif (t_roots is not None) and (y_roots is not None):
+            if np.isscalar(t_roots):
+                print('%6.1f %15.4f %15.2f' % (t_roots, y_roots[0], y_roots[1]))
+            else:
+                for (t, y, v) in zip(t_roots, y_roots[:, 0], y_roots[:, 1]):
+                    print('%6.1f %15.4f %15.2f' % (t, y, v))
+        else:
+            print('Error: one of (t_roots, y_roots) is None while the other not.')
+    else:
+        print('Computation failed.')
+
+# Set tspan to end at t_end1
+tspan = np.arange(0, t_end1 + 1, 1.0, np.float)
+#
+# 1. Solve the problem without interrupt function i.e. compute until
+#    the first root is found
+#
+solver = cvode.CVODE(rhs_fn, nr_rootfns=1, rootfn=root_fn, old_api=False)
+print_results(1, solver.solve(tspan, y0))
+
+#
+# 2. Solve the problem with interrupt function i.e. compute until
+#    the final time is reached
+#
+solver = cvode.CVODE(rhs_fn, nr_rootfns=1, rootfn=root_fn, onroot=onroot_va, old_api=False)
+print_results(2, solver.solve(tspan, y0), require_no_roots=True)
+
+# Set tspan to end at t_end2
+tspan = np.arange(0, t_end2 + 1, 1.0, np.float)
+#
+# 3. Solve the problem without interrupt function i.e. compute until
+#    the first root is found
+#
+solver = cvode.CVODE(rhs_fn, nr_rootfns=1, rootfn=root_fn, old_api=False)
+print_results(3, solver.solve(tspan, y0))
+
+#
+# 4. Solve the problem with interrupt function i.e. compute until
+#    the final time is reached
+#
+solver = cvode.CVODE(rhs_fn, nr_rootfns=1, rootfn=root_fn, onroot=onroot_va, old_api=False)
+print_results(4, solver.solve(tspan, y0))
+
+#
+# 5. Solve the problem without interrupt function i.e. compute until
+#    the first root is found
+#
+solver = cvode.CVODE(rhs_fn, nr_rootfns=1, rootfn=root_fn, onroot=onroot_vb, old_api=False)
+print_results(5, solver.solve(tspan, y0))
+
+#
+# 6. Solve the problem with interrupt function i.e. compute until
+#    the final time is reached
```

Comment:
this uses onroot_vc, so it stops at first root after t=28, and hence need not go to the final time